TipCatDotNet.Api/Services/HospitalityFacilities/FacilityService.cs

```diff
@@ -18,13 +23,54 @@ public FacilityService(ILoggerFactory loggerFactory, AetherDbContext context)
         }
 
 
+        public Task<Result<FacilityResponse>> Add(MemberContext memberContext, FacilityRequest request, CancellationToken cancellationToken)
+        {
+            return Result.Success()
+                .EnsureCurrentMemberBelongsToAccount(memberContext.AccountId, request.AccountId)
+                .Bind(AddInternal)
+                .Bind(facilityId => GetFacility(facilityId, cancellationToken));
+
+
+            // Result Validate()
+            // {
+            //     var validator = new FacilityRequestAddValidator();
+            //     var validationResult = validator.Validate(request);
+            //     if (!validationResult.IsValid)
+            //         return validationResult.ToFailureResult();
+
+            //     return Result.Success();
+            // }
+
+
+            async Task<Result<int>> AddInternal()
+            {
+                var now = DateTime.UtcNow;
+
+                var newFacility = new Facility
+                {
+                    Name = request.Name,
+                    AccountId = request.AccountId.Value,
```

Comment:
At this point `AccountId` is definitely not null. Could it be like `request.AccountId!`?